File: Shortest_Unsorted_Continuous_Subarray/findUnsortedSubarray.c
#include<stdio.h>

#define MIN(a,b) (((a)<(b))?(a):(b))
#define MAX(a,b) (((a)>(b))?(a):(b))

int findUnsortedSubarray(int *nums, int size) {
    int i, min = INT_MAX, max = INT_MIN;
    
    for (i = 0; i < size; i++) {
        if (nums[i] < min) {
            min = MIN(min, nums[i]);
        }
        if (nums[i] > max) {
            max = MAX(max, nums[i]);
        }
    }
    
    int l, r;
    for (l = 0; l < size; l++) {
        if (min < *(nums+l))
            break;
    }
    for (r = size-1; r >= 0; r--) {
        if (max > *(nums+r))
            break;
    }
    
    return r-l < 0 ? 0 : r-l+1;
}

int main(void) {
    int nums[7] = {2, 6, 4, 8, 10, 9, 15};
    
    printf("%d\n", findUnsortedSubarray(&nums, sizeof(nums)/sizeof(nums[0])));
    return 0;
}

File: Shortest_Unsorted_Continuous_Subarray/FindUnsortedSubarray.java
class Solution {
    public int FindUnsortedSubarray(int[] nums) {
        int n = nums.length, begin = -1, end = -2, min = Integer.MAX_VALUE, max = Integer.MIN_VALUE;
        
        for (int i = 0; i < n; i++) {
            max = Math.max(max, nums[i]);
            min = Math.min(min, nums[n-1-i]);
            if (nums[i] < max) end = i;
            if (nums[n-1-i] > min) begin = n-1-i;
        }

        return end - begin + 1;
    }

    public static void main(String[] args) {
        int[] nums = {2, 6, 4, 8, 10, 9, 15};
        Solution solution = new Solution();
        System.out.println(solution.FindUnsortedSubarray(nums));
    }
}
